custom_components/sscpoe/config_flow.py

```diff
@@ -145,19 +166,281 @@ def login():
+                            
+                            # Extract model from first 10 characters of SN
+                            if sn and len(sn) >= 10:
+                                model_prefix = sn[:10]
+                                model = model_prefix.rstrip('0')  # Remove trailing zeros
+                                if len(model) >= 3:  # Ensure minimum length
+                                    pass  # model is already valid
+                                else:
+                                    model = "Unknown"
+                            else:
+                                model = "Unknown"
+                                
+                            # Firmware version
+                            firmware = calldata.get("V", "Unknown")
+                        else:
+                            sn = f"web_{ip.replace('.', '_')}"
+                            model = "Unknown"
+                            firmware = "Unknown"
+                    except Exception as e:
+                        LOGGER.debug("Failed to get device info for %s: %s", ip, e)
+                        sn = f"web_{ip.replace('.', '_')}"
+                        model = "Unknown"
+                        firmware = "Unknown"
+
+                    # Build single-item list using unified logic
+                    self.local_devices = [
+                        self._normalize_device_data({
+                            "ip": ip,
+                            "sn": sn,
+                            "V": firmware,
+                            "Active_state": "active",
+                            "web_token": device["web_token"],
+                            "add_method": "web",
+                            "user_password": password
+                        })
+                    ]
+                    return await self.async_step_local_select()
+    
+        return self.async_show_form(
+            step_id="local_web",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_IP_ADDRESS, default=ip): str,
+                    vol.Required(CONF_PASSWORD, default=password): str,
+                }
+            ),
+            errors=errors,
+        )
+    
+    async def async_step_local_multicast(self, user_input: dict[str, Any] = None) -> FlowResult:
+        """Screen 2M: multicast discovery settings + scan, then go to local confirm screen."""
+        errors: dict[str, str] = {}
+    
+        # Get list of interfaces with their IP addresses
+        iface_choices = self._get_interface_choices()
+        if not iface_choices:
+            errors["base"] = "no_interfaces_with_ip"
+            return self.async_show_form(
+                step_id="local_multicast",
+                data_schema=vol.Schema({}),
+                errors=errors,
+            )        
+        
+        bind_default = DEFAULT_BIND_INTERFACE
+        set_ttl_default = False
+        ttl_default = 2
```

Comment:
```suggestion
        DEFAULT_TTL = 2
```
and move to the file beginning.